container.sh

```diff
@@ -0,0 +1,26 @@
+ENGINE=docker
+
+if ! which "$ENGINE" &>/dev/null ; then
+    ENGINE=podman
+fi
+
+if ! which "$ENGINE" &>/dev/null ; then
+    echo "No docker or podman in PATH" >&2
+    exit 1
+fi
+
+read -d '' script <<EOF
+cd /app/nix-user-chroot
+mkdir .nix
+cargo build
+
+run() {
+    cargo run .nix "\$@"
+}
+
+export -f run
+exec bash
+EOF
+
+# "$ENGINE" run -v .:/app -it rust bash -c "$script"
+"$ENGINE" run -v ..:/app -it rust bash -c "$script"
```

Comment:
That does not quite work with docker:

```suggestion
"$ENGINE" run -v $PWD:/app -it rust bash -c "$script"
```

Reply:
Hehe, my bad, that was left from when I tried to use rooter so I used the parent directory which contained both :)

Reply:
maybe this one is safer:

```
DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```

Reply:
Sure, sounds about right

Reply:
I just added the cd, we don't really need the user's cd elsewhere